src/object/srv_obj_migrate.c

```diff
@@ -3020,7 +3020,7 @@ migrate_one_object(daos_unit_oid_t oid, daos_epoch_t eph, daos_epoch_t punched_e
 		 * pseudorandomly to get better performance by leveraging all
 		 * the xstreams.
 		 */
-		ult_tgt_idx = oid.id_pub.lo % dss_tgt_nr;
+		ult_tgt_idx = d_rand() % dss_tgt_nr;
```

Comment:
if we just want to choose ULT evenly, you probably can have
unsigned int ult_tgt_selector;
......
ult_tgt_idx = (ult_tgt_selector ++) % dss_tgt_nr;